src/actions/mod.rs

```diff
@@ -197,21 +197,81 @@ impl InitActionContext {
     }
 
     fn build<O: Output>(&self, project_path: &Path, priority: BuildPriority, out: O) {
-        struct BuildNotifier<O: Output> {
+
+        lazy_static! {
+            static ref PROGRESS_ID_COUNTER: AtomicUsize = {
```

Comment:
I think the counter, along with traits implementations using it would benefit from separating into a common file `src/actions/progress.rs`?